src/lightning_app/runners/cloud.py

```diff
@@ -248,36 +332,50 @@ def load_app_from_file(cls, filepath: str) -> "LightningApp":
             del os.environ["LAI_RUNNING_IN_CLOUD"]
         return app
 
-    def _resolve_config(self, name: Optional[str]) -> AppConfig:
+    def _resolve_config(self, name: Optional[str], load: bool = True) -> AppConfig:
         """Find and load the config file if it exists (otherwise create an empty config).
 
         Override the name if provided.
         """
-        config_file = _get_config_file(self.entrypoint_file)
-        cloudspace_config = AppConfig.load_from_file(config_file) if config_file.exists() else AppConfig()
+        config_file = _get_config_file(self.entrypoint)
+        cloudspace_config = AppConfig.load_from_file(config_file) if config_file.exists() and load else AppConfig()
         if name:
             # Override the name if provided
             cloudspace_config.name = name
         return cloudspace_config
 
     def _resolve_root(self) -> Path:
         """Determine the root of the project."""
-        return Path(self.entrypoint_file).absolute().parent
-
-    def _resolve_repo(self, root: Path) -> LocalSourceCodeDir:
+        root = Path(self.entrypoint).absolute()
+        if root.is_file():
+            root = root.parent
+        return root
+
+    def _resolve_open_ignore_functions(self) -> List[_IGNORE_FUNCTION]:
+        entrypoint = self.entrypoint.absolute()
+        if entrypoint.is_file():
+            return [lambda src, paths: [path for path in paths if path.absolute() == entrypoint]]
+        return []
```

Comment:
I think this needs a docstring. returning random lambda functions in this way isn't totally clear

Reply:
Added